numba/core/bytecode.py

```diff
@@ -168,6 +198,16 @@ def _unpack_opargs(code):
         else:
             arg = None
             i += NO_ARG_LEN
+            if PYVERSION in ((3, 12), ):
+                # Python 3.12 introduced cache slots. We need to account for
+                # cache slots when we determine the offset of the next opcode.
+                # The number of cache slots is specific to each opcode and can
+                # be looked up in the _inline_cache_entries dictionary.
+                i += _inline_cache_entries[op] * 2
```

Comment:
Same question about `2` being a named constant.

Reply:
Fixed in 9353b9a20d040d737381bdb42f22a40a9c9e46e0

Reply:
and in 101fcfb57f5b005932d7e04bd4eed33a728f0058